python 3.10 collections test suite update

mburridge96 opened the pull request:
Python 3.10 collections depreciated the use of Iterable. Iterable has now been moved to collections.abc.

Tested in python 3.9 and 3.10 and passed 100% of tests. Python 3.8 had type issues with SBOL.components (ABCMeta type not subscriptable) so could not run helper functions. Question is does the testing suite require python 3.8 support? If it does will have to look into it further  

Updated requirements.txt to be more exhaustive of requirements. 

Added small troubleshooting section for Ubuntu setup.
fix #310

mburridge96 committed:
Added .vscode to gitignore
Updated requirements.txt to include all required packages
Collections Iterable depreciated, now located in collections.abs

mburridge96 committed:
Added .vscode to gitignore
Updated requirements.txt to include all required packages
Collections Iterable depreciated, now located in collections.abs

mburridge96 committed:
Added troubleshooting section to readme.
Further updated requirements.txt from successful passing test build

mburridge96 committed:
Added troubleshooting section to readme.
Further updated requirements.txt from successful passing test build

mburridge96 committed:
Specified ubuntu troubleshooting

mburridge96 committed:
Specified ubuntu troubleshooting

jakebeal commented on scripts/requirements.txt:
```diff
@@ -1,6 +1,45 @@
-sbol3>=1.0b10
-sbol-utilities>=1.0a15
-biopython
+astroid==2.11.7
```
It's typically better to avoid including indirect requirements in the requirements.txt list.

mburridge96 replied:
Ah ok didn't realise! Always thought it was better to be more explicit to have a more consistent environment. Will remove.

jakebeal commented on scripts/scriptutils/helpers.py:
```diff
@@ -1,5 +1,5 @@
 import math
-from collections import Iterable
+from collections.abc import Iterable
```
Is `iterable` a type options? If so, we can get backward compatibility through `from __future__ import annotations`

mburridge96 replied:
So using __future__ does allow collections to be backwards compatible, but v3.8 didn't yet have .removesuffix() for strings which is used through part_retrieve.py. 

Added some handling to this to get around this issue, but 2 tests still fail in py v3.8 as filecmp asserts deem integrated package and collated file to be different   in test_collation  and  test_all_import_types.

jakebeal replied:
Those issues are likely due to the update in rdflib that causes it to no longer have a spurious newline, which I fixed in #311. If you update your branch, it should resolve those issues.  If it doesn't resolve those issues, it will be necessary to look deeper.

mburridge96 committed:
Added __future__ to handle collections backwards compatibility.
Added handling for py v3.8 prefix/suffix handling.
py v3.8 still fails on test_collation adn test_all_import_types

mburridge96 committed:
Added __future__ to handle collections backwards compatibility.
Added handling for py v3.8 prefix/suffix handling.
py v3.8 still fails on test_collation adn test_all_import_types

mburridge96 committed:
Revert back to direct requirements

mburridge96 committed:
Revert back to direct requirements

mburridge96 committed:
Merge branch 'develop' of https://github.com/mburridge96/iGEM-distribution into develop

mburridge96 committed:
Merge branch 'iGEM-Engineering:develop' into develop

jakebeal commented:
@mburridge96 Are these failures expected here?

mburridge96 commented:
Not at all, should have failed on testing but not on linting. Will have a look

mburridge96 committed:
Updated actions.
Deleted synchronise and updated with actions based build_packages

mburridge96 committed:
Updated part_retrieval formating for flake8

mburridge96 committed:
Updated paths.
Updated setup-python v2 -> v4.
Updated checkout v2 -> v3

mburridge96 committed:
checking out repo within workflow

mburridge96 committed:
reverted pythonl_linter due to not running

mburridge96 committed:
format updated for flake8

mburridge96 committed:
format updated for flake8

mburridge96 committed:
format updated for flake8
